core/src/main/java/org/apache/struts2/interceptor/I18nInterceptor.java

```diff
@@ -282,14 +282,15 @@ public Locale find() {
         @Override
         public Locale store(ActionInvocation invocation, Locale locale) {
             HttpSession session = ServletActionContext.getRequest().getSession(false);
+            Map<String, Object> invocationSession = invocation.getInvocationContext().getSession();
 
-            if (session != null) {
+            if (session != null && invocationSession != null) {
```

Comment:
I'm not sure if this is the best approach, `invocationSession` cannot be normally `null`, it's always initialised with `Dispatcher`. It can be empty but not `null`. I would rather solve this by initialising a test environment properly.

Reply:
:+1: thanks! then I think a similar fix like [https://issues.apache.org/jira/browse/WW-3442](WW-3442) seems good.

Reply:
Yes and know, the my fix had introduced a regression and I wonder if we can fix that without introducing the above change.

Reply:
> the my fix

Did you mean WW-3442? Or WW-4741?

Reply:
WW-4741

Reply:
That is a good fix :+1: I thought when httpSession is null, then invocationSession also will be null but as you corrected me, it won't be but will be empty. So this PR isn't valid anymore and I like your fix instead :)